Compare/compareroot.cpp

```diff
@@ -103,7 +103,8 @@ int main(int argc,char *argv[]){
 	      if(size1 || size2) found_non_empty_vector = true;
 	      if(size1 == size2) {
 		for(size_t j = 0; j < size1; j++) {
-		  same *= vec1->at(j) == vec2->at(j);
+		  //same *= vec1->at(j) == vec2->at(j); //In my experience, comparing doubles like this can cause problems.
+		  same *= std::fabs(vec1->at(j)-vec2->at(j)) < 1e-8; //Comparing like this avoids precision problems (Is this precision okay?)
```

Comment:
This is a good point!

(Writing this down as I had a quick think!)
My first thought was something like checking for `std::fabs( vec1->at(j)-vec2->at(j) ) <= std::min( 1e-8 * std::max( std::fabs( vec1->at(j) ), std::fabs( vec1->at(j) ) ), 1e-8 )`

I think this effectively ensures that the tolerance is at least 1e-8, but it could be higher depending on the magnitude of hit_time_1 and hit_time_2, it essentially shifts it by the power of ten...

So if 1e-8 * max_value >= 1e-8 ; the tolerance is always 1e-8.

if 1e-8 * max_value < 1e-8 ; the tolerence is shifted by the power of 10.

Does that seem sensible?

Reply:
Seems like it could work!

I'm currently [going down a rabbit hole](https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/) if you'd like to join me 🕳️